cmd/project/samples.go

```diff
@@ -55,8 +55,13 @@ func NewSamplesCommand(clients *shared.ClientFactory) *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVarP(&samplesTemplateURLFlag, "template", "t", "", "template URL for your app")
+	// DEPRECATED(semver:major): Prefer the create command when repository details are known
 	cmd.Flags().StringVarP(&samplesGitBranchFlag, "branch", "b", "", "name of git branch to checkout")
+	cmd.Flag("branch").Hidden = true
```

Comment:
thought: FWIW I think `--branch` has a legit use-case with the `slack samples` command. However, I get the idea that `slack samples` is meant to discover the available templates and isn't meant to target advanced use-cases. The `slack create` command is designed to handle the general-purpose and advanced ones.

Reply:
@mwbrooks Thanks for sharing this - I was thinking the same to recommend the `create` command when creating a specific branch.

I'm wondering if we can revisit this in following change to show samples that have an existing branch when this flag is provided to avoid errors during a `git` clone?

Will be thinking about this and related docs! 📚